src/CardInfo.js

```diff
@@ -3,11 +3,73 @@ import React from 'react';
 import './App.css';
 
 export class CardInfo extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isEditing: false,
+            oldDescription: props.description,
+            description: props.description
+        }
+    }
+
+    handleChangeButton = (event) => {
+        this.setState({
+            isEditing: true
+        });
+    }
+
+    handleSaveButton = async(event) => {
+        const {onChangeCard} = this.props;
+
+        let body = {
+            description: this.state.description
+        }        
+        
+        await onChangeCard(body, this.props);       
+
+        this.setState({
+            oldDescription: this.state.description,
+            isEditing: false
+        });
+    }
+
+    handleCancelButton = (event) => {
+        this.setState({
+            description: this.state.oldDescription,
```

Comment:
why can't you just use `this.props.description` to restore value?

Reply:
> why can't you just use `this.props.description` to restore value?


I can’t just use props.description for recovery because the value that was with render () will be stored there.
If I edit description several times, the value in props will be irrelevant.